feat(update): configurable cert update command

baby-gnu opened the pull request:
<!--
Please fill in this PR template to make it easier to review and merge.

It has been designed so that a lot of it can be completed *after* submitting it,
e.g. filling in the checklists.

Notes:
1. Please keep the PR as small as is practicable; the larger a PR gets, the harder it becomes to review and the more time it requires to get it merged.
2. Similarly, please avoid PRs that cover more than one type; it should be a _bug fix_ *OR* a _new feature_ *OR* a _refactor_, etc.
3. Please direct questions to the [`#formulas` channel on Slack](https://saltstackcommunity.slack.com/messages/C7LG8SV54/), which is bridged to `#saltstack-formulas` on Freenode.
4. Feel free to suggest improvements to this template by reporting an issue or submitting a PR.  The source of this template is:
  - https://github.com/saltstack-formulas/.github/blob/master/.github/pull_request_template.md
-->

### PR progress checklist (to be filled in by reviewers)
<!-- Please leave this checklist for reviewers to tick as they work through the PR. -->

- [ ] Changes to documentation are appropriate (or tick if not required)
- [ ] Changes to tests are appropriate (or tick if not required)
- [ ] Reviews completed

---

### What type of PR is this?
<!-- Please tick each box that is relevant (after creating the PR). -->

#### Primary type
<!-- There really should be only *one* of these types ticked for each PR. -->

- [ ] `[build]`    Changes related to the build system
- [ ] `[chore]`    Changes to the build process or auxiliary tools and libraries such as documentation generation
- [ ] `[ci]`       Changes to the continuous integration configuration
- [x] `[feat]`     A new feature
- [ ] `[fix]`      A bug fix
- [ ] `[perf]`     A code change that improves performance
- [ ] `[refactor]` A code change that neither fixes a bug nor adds a feature
- [ ] `[revert]`   A change used to revert a previous commit
- [ ] `[style]`    Changes that do not affect the meaning of the code (white-space, formatting, missing semi-colons, etc.)

#### Secondary type
<!-- Most PRs should include all of the following types as well. -->

- [x] `[docs]`     Documentation changes
- [ ] `[test]`     Adding missing or correcting existing tests

### Does this PR introduce a `BREAKING CHANGE`?
<!-- If so, change the following to a `Yes` and explain what the breaking changes are. -->
<!-- If there are multiple breaking changes, list them all. -->

No.

### Related issues and/or pull requests
<!-- Please link any related issues/PRs here, especially any issues that are closed by this PR. -->

#26

### Describe the changes you're proposing
<!-- A clear and concise description of what you have implemented. -->
<!-- Consider explaining each commit if they cover different aspects of the proposed changes. -->

Make integration of the certificates in system store configurable:

- several systems seems to be able to use the `update-ca-certificates` 
- RedHat based system has another system (`ca-trust`) that will require modifications in how this formula manage certificates (put them under `/etc/ca-trust/source/`)

This PR only makes configurable the command to update of system store:

- only Debian familly will use it like done previously
- a dedicated `.sls` permits to only use [`onchanges` requisite](https://docs.saltproject.io/en/latest/ref/states/requisites.html#requisites-onchanges) to detect modification

### Pillar / config required to test the proposed changes
<!-- Provide links to the SLS files and/or relevant configs (be sure to remove sensitive info). -->



### Debug log showing how the proposed changes work
<!-- Include a debug log showing how these changes work, e.g. using `salt-minion -l debug`. -->
<!-- Alternatively, linking to Kitchen debug logs is useful, e.g. via. Travis CI. -->
<!-- Most useful is providing a passing InSpec test, which can be used to verify any proposed changes. -->



### Documentation checklist
<!-- Please tick each box that is relevant (after creating the PR). -->

- [x] Updated the `README` (e.g. `Available states`).
- [ ] Updated `pillar.example`.

### Testing checklist
<!-- Please tick each box that is relevant (after creating the PR). -->

- [x] Included in Kitchen (i.e. under `state_top`).
- [ ] Covered by new/existing tests (e.g. InSpec, Serverspec, etc.).
- [ ] Updated the relevant test pillar.

### Additional context
<!-- Add any other context about the proposed changes here. -->

daks commented on cert/parameters/defaults.yaml:
```diff
@@ -17,5 +17,6 @@ values:
   cert_mode: 644
   key_mode: 600
   remove: false
+  certs_update_cmd: ~
```
is it needed to insert this key with an empty value _(IIRC `~` is used to 'clean' a k/v in yaml_) to be able to use `{%- if mapdata.certs_update_cmd %}`? I would have thought that putting no value would be sufficient.

baby-gnu replied:
I'll test tomorrow but I think `yamllint` will not like having a key without value.

myii replied:
> I'll test tomorrow but I think `yamllint` will not like having a key without value.

@baby-gnu I'm a little wary of using `null`/`~` in YAML as well, since it doesn't always have the desired effect -- I would be more interested in it if it actually _unset_ the value altogether.  Anyway, wouldn't this work?

```suggestion
  certs_update_cmd: ""
```

baby-gnu replied:
I did that, thanks.

daks commented on test/integration/default/files/_mapdata/amazonlinux-2.yaml:
```diff
@@ -28,6 +28,7 @@ values:
     cert.from.src.to.add: {}
     cert.to.add:
       cert: "-----BEGIN CERTIFICATE-----\n2MOCKED CERT\n-----END CERTIFICATE----- "
+  certs_update_cmd: null
```
here too, are those `null` value absolutely needed, isn't the default value sufficient?

baby-gnu replied:
This is the files needed to check what `map.jinja` load, it must reflect the content of `cert._mapdata`.

daks reviewed (approved):
Except for the two minor comments, all looks good to me, and I like the solution you implemented instead of the previous `if` one _(which I don't like generally)_.
Thanks!

baby-gnu committed:
test(inspec): integrade update command in mapdata dump files

The update of system wide certificate bundle is only supported on
Debian family.

baby-gnu committed:
feat(update): integrate certificates on supported systems

The update of system wide certificate bundle is only supported on
Debian family for now.

The split of the update to a dedicated `.sls` file permits to only use
`onchanges` requisite.

saltstack-formulas-travis commented:
:tada: This PR is included in version 1.3.0 :tada:

The release is available on [GitHub release](https://github.com/saltstack-formulas/cert-formula/releases/tag/v1.3.0)

Your **[semantic-release](https://github.com/semantic-release/semantic-release)** bot :package::rocket:
